src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs

```diff
@@ -245,21 +232,40 @@ private static bool TryIsMethodGetParametersUnsupported(MethodInfo method, [NotN
             return null;
         }
 
-        sb.Append(MapType(method.ReturnType));
-        sb.Append($" {_fixupSymbolName(pinvoke.EntryPoint)} (");
-        int pindex = 0;
-        var pars = method.GetParameters();
-        foreach (var p in pars)
-        {
-            if (pindex > 0)
-                sb.Append(',');
-            sb.Append(MapType(pars[pindex].ParameterType));
-            pindex++;
+        if (pinvoke.WasmLinkage) {
+            sb.Append($"__attribute__((import_module(\"{EscapeLiteral(pinvoke.Module)}\"),import_name(\"{EscapeLiteral(pinvoke.EntryPoint)}\")))\nextern ");
         }
-        sb.Append(");");
+
+        sb.Append($"{MapType(method.ReturnType)} {CEntryPoint(pinvoke)} ({string.Join(", ", method.GetParameters().Select(p => MapType(p.ParameterType)))});");
+
         return sb.ToString();
     }
 
+    private string CEntryPoint(PInvokeCallback export)
+    {
+        if (export.EntryName is not null) {
+            return _fixupSymbolName(export.EntryName);
+        }
+
+        var method = export.Method;
+        // EntryPoint wasn't specified generate a name for the entry point
+        return _fixupSymbolName($"wasm_native_to_interp_{method.DeclaringType!.Module!.Assembly!.GetName()!.Name!}_{method.DeclaringType.Name}_{method.Name}");
+    }
+
+    private string DelegateKey(PInvokeCallback export)
+    {
+        // FIXME: this is a hack, we need to encode this better
+        // and allow reflection in the interp case but either way
+        // it needs to match the key passed to get_native_to_interp
```

Comment:
Need to open an issue about this.  It's ugly and complicating a fix for calling into the entry point without having to create a managed delegate to initialize the lookup table